duckduckgo/Android
# Add alternative onboarding for interstitial search users

Task/Issue URL: https://app.asana.com/0/inbox/361428290920650/1163294449813158/1165440117435986

**Description**:
Users may experience our app initially via our new interstitial search screen by launching the widget or future system search box and thus skip our initial in-app onboarding. This PR adds an alternative onboarding experience for these users.

**Steps to test this PR**:

**Test new onboarding:**
1. Install the app cleanly
1. Install the widget
1. Launch the app from the widget
1. Note that the interstitial onboarding is displayed and the "m_io_s" pixel fires
1. Press "Show More" and note that more info is displayed and the "m_io_m" pixel fires
1. Press "Hide" and note that less info is displayed and the "m_io_l" pixel fires
1. Press "Got it" and note that the onboarding disappears and the "m_io_d" pixel fires
1. Launch the app from the widget again and note that the onboarding does not appear and none of the above pixels fire
1. Launch the app from the app icon
1. Note that the app is launched WITHOUT the initial onboarding


**Test that completing in-app onboarding suppresses interstitial onboarding:**
1. Install the app cleanly
1. Launch the app from the app icon and complete the initial onboarding
1. Install the widget
1. Launch the app from the widget
1. Note that the interstitial screen is launched WITHOUT any onboarding

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
